model/account_metadata.go

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+//
+
+package model
+
+import "encoding/json"
+
+// AccountMetadata is the provider metadata stored in a model.Account.
+type AccountMetadata struct {
+	Provision bool
```

Comment:
No basically this stores whether the account is provisioned or not. Because there may be cases that we create an empty account without provisioning anything inside. Makes sense?